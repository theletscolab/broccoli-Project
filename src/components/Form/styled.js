import styled, { css } from 'styled-components/macro'

const errorStyles = ({ error, theme }) =>
  error && {
    borderColor: theme.colors.error,
    color: theme.colors.error
  }

const FormStyled = styled.form`
  display: block;
  padding: 3.2rem 2.4rem;
  width: 300px;
  ${({ theme }) => css`
    ${theme.mediaQueries.md} {
      width: 375px;
    }
  `};
`
const FormTitle = styled.h2`
  display: block;
  color: ${({ theme }) => theme.colors.grey900};
  font-size: ${({ theme }) => theme.fontSizes[2]};
  font-weight: ${({ theme }) => theme.fontWeights.bold};
  line-height: 1.31;
  margin-bottom: 3.6rem;
  position: relative;

  &::after {
    background-color: ${({ theme }) => theme.colors.grey400};
    bottom: -12px;
    content: ' ';
    height: 2px;
    left: calc(50% - 25px);
    position: absolute;
    width: 50px;
  }
`

const ErrorMessage = styled.div`
  color: ${({ theme }) => theme.colors.error};
  font-size: ${({ theme }) => theme.fontSizes[0]};
  font-weight: ${({ theme }) => theme.fontWeights.medium};
  padding: 0.6rem;
  margin-top: 1rem;
  margin-bottom: 1rem;
  text-align: right;
`
const InputStyled = styled.input`
  background-color: ${({ theme }) => theme.colors.grey50};
  border: 1px solid ${({ theme }) => theme.colors.grey700};
  border-radius: 1;
  color: ${({ theme }) => theme.colors.grey900};
  display: block;
  font-size: 1.4rem;
  height: 48px;

  padding: 1.2rem 1.6rem;
  width: 100%;

  ${errorStyles}
`

export { FormStyled, FormTitle, ErrorMessage, InputStyled }
