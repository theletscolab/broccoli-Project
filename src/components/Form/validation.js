import { yupResolver } from '@hookform/resolvers/yup'
import * as Yup from 'yup'

// form validation rules
const validationSchema = Yup.object().shape({
  name: Yup.string()
    .required('Name is required')
    .matches(
      /(^[a-zA-Z]{3,}$|^[a-zA-Z]{3,}\s[a-zA-Z]{1,}$)/,
      'Full name need to be at least 3 characters'
    ),
  email: Yup.string().required('Email is required').email('Email is invalid'),
  confirmEmail: Yup.string()
    .oneOf([Yup.ref('email'), null], 'Email must match')
    .required('Confirm Email is required')
})
export const formOptions = { resolver: yupResolver(validationSchema) }
