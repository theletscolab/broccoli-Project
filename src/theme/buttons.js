export const buttonSizes = {
  small: {
    height: '42px',
    fontSize: '1.4rem',
    padding: '1.2rem 1.7rem'
  },
  medium: {
    height: '48px',
    fontSize: '1.6rem',
    padding: '1.4rem 2.4rem'
  },
  large: {
    height: '54px',
    fontSize: '1.8rem',
    padding: '1.6rem 2.8rem'
  }
}
