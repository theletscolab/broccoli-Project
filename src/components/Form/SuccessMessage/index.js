import React from 'react'
import PropTypes from 'prop-types'
import { ModalContext } from '../../../contexts/modalContext'
import Button from '../../Button'
import { StyledSuccessMessage, MessageTitle, Message } from './styled'

const SuccessMessage = ({ formTitle, formMessage, formBtnTxt }) => {
  let { handleModal } = React.useContext(ModalContext)
  return (
    <StyledSuccessMessage>
      <MessageTitle>{formTitle}</MessageTitle>
      <Message>{formMessage}</Message>
      <Button title={formBtnTxt} onClick={() => handleModal()} />
    </StyledSuccessMessage>
  )
}

SuccessMessage.propTypes = {
  formTitle: PropTypes.string.isRequired,
  formMessage: PropTypes.string.isRequired,
  formBtnTxt: PropTypes.string.isRequired
}

SuccessMessage.defaultProps = {
  formTitle: 'All done!',
  formMessage: 'You will be one of the first to experience Broccoli & Co. when we launch.',
  formBtnTxt: 'OK'
}

export default SuccessMessage
