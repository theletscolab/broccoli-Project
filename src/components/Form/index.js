import React from 'react'
import PropTypes from 'prop-types'
import { useForm } from 'react-hook-form'
import { postForm } from '../../features/api'
import Button from '../../components/Button'
import { FormStyled, FormTitle, ErrorMessage, InputStyled } from './styled'
import { formOptions } from './validation'

const Form = ({ formTitle, formBtnTxt, onSuccess }) => {
  const {
    register,
    formState: { errors, isSubmitting },
    handleSubmit
  } = useForm(formOptions)

  const onSubmit = async data => ((await postForm(data)) ? onSuccess() : null)

  return (
    <FormStyled onSubmit={handleSubmit(onSubmit)}>
      <FormTitle>{formTitle}</FormTitle>
      <InputStyled
        name="name"
        type="text"
        placeholder="Full name"
        {...register('name')}
        className={`form-control ${errors.name ? 'is-invalid' : ''}`}
      />
      <ErrorMessage> {errors.name?.message} </ErrorMessage>
      <InputStyled
        name="email"
        type="email"
        placeholder="Email"
        {...register('email')}
        className={`form-control ${errors.email ? 'is-invalid' : ''}`}
      />
      <ErrorMessage> {errors.email?.message} </ErrorMessage>
      <InputStyled
        name="confirmEmail"
        type="email"
        placeholder="Confirm email"
        {...register('confirmEmail')}
        className={`form-control ${errors.confirmEmail ? 'is-invalid' : ''}`}
      />
      <ErrorMessage> {errors.confirmEmail?.message} </ErrorMessage>
      {!isSubmitting ? (
        <Button type="submit" title={formBtnTxt} />
      ) : (
        <Button type="submit" disabled={isSubmitting} title={'Submitting'} />
      )}
    </FormStyled>
  )
}

Form.propTypes = {
  formTitle: PropTypes.string,
  formBtnTxt: PropTypes.string,
  onSuccess: PropTypes.func.isRequired
}

Form.defaultProps = {
  formTitle: 'Request an invite',
  formBtnTxt: 'Send'
}

export default Form
