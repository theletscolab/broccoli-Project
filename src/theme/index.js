import { buttonSizes } from './buttons'
// App theme

// Colors Palette
const awlxDarkBlue = '#612fff'
const awlxLightBlue = '#e9e3ff'
const white = '#ffffff'
const grey50 = '#fafafa'
const grey100 = '#f5f5f5'
const grey200 = '#eeeeee'
const grey300 = '#e0e0e0'
const grey400 = '#bdbdbd'
const grey500 = '#9e9e9e'
const grey600 = '#757575'
const grey700 = '#616161'
const grey800 = '#424242'
const grey900 = '#212121'
const black = '#000000'
const black15 = 'rgba(0, 0, 0, 0.15)'
const black25 = 'rgba(0, 0, 0, 0.25)'
const black50 = 'rgba(0, 0, 0, 0.50)'
const black75 = 'rgba(0, 0, 0, 0.75)'
const error = '#cc0000'
const bodyBg = white
const bodyColor = grey900

// Responsive breakpoints and media-queries (mobile first).
const breakpoints = ['576px', '768px', '992px', '1200px']

const fonts = {
  base: `'Montserrat', sans-serif`,
  caveat: `'Caveat', cursive`
}

const fontWeights = {
  light: 300,
  regular: 400,
  medium: 500,
  semiBold: 600,
  bold: 700
}

const headingFontFamily = fonts.base

const fontSizes = ['1.2rem', '1.6rem', '2rem', '2.4rem', '3.2rem', '4.8rem', '6.4rem']

const bodyText = fontSizes[1]
const headingWeight = fontWeights.bold
const headingLineHeight = 1.4

const theme = {
  breakpoints,
  mediaQueries: {
    sm: `@media screen and (min-width: ${breakpoints[0]})`,
    md: `@media screen and (min-width: ${breakpoints[1]})`,
    lg: `@media screen and (min-width: ${breakpoints[2]})`,
    xl: `@media screen and (min-width: ${breakpoints[3]})`
  },
  colors: {
    awlxDarkBlue,
    awlxLightBlue,
    white,
    grey50,
    grey100,
    grey200,
    grey300,
    grey400,
    grey500,
    grey600,
    grey700,
    grey800,
    grey900,
    black,
    black15,
    black25,
    black50,
    black75,
    bodyBg,
    bodyColor,
    error
  },
  space: [0, 4, 8, 16, 32, 64, 128, 256],
  fonts,
  fontSizes,
  bodyText,
  fontWeights,
  headingWeight,
  headingLineHeight,
  headingFontFamily,
  shadows: {
    small: '0 0 4px rgba(0, 0, 0, .125)',
    large: '0 0 24px rgba(0, 0, 0, .125)'
  },
  maxContainerWidth: '992px',
  buttonSizes
}

export default theme
