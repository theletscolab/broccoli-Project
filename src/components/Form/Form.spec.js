import { render, cleanup } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import Form from './index'

describe('components > Form', () => {
  afterEach(cleanup)
  let component

  describe('with defaults', () => {
    const onSucess = jest.fn()
    beforeAll(() => {
      component = render(<Form onSuccess={onSucess} />, {
        wrapper: ({ children }) => <ThemeProvider theme={theme}> {children}</ThemeProvider>
      })
    })

    it('rendered without crashing', () => {
      expect(component.asFragment()).toMatchSnapshot()
    })
  })
})
