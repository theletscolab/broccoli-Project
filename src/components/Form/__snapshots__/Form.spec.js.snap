// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > Form with defaults rendered without crashing 1`] = `
<DocumentFragment>
   
  <form
    class="styled__FormStyled-sc-133yddn-0 dCGduS"
  >
    <h2
      class="styled__FormTitle-sc-133yddn-1 caCceO"
    >
      Request an invite
    </h2>
    <input
      class="styled__InputStyled-sc-133yddn-3 QcXMb form-control "
      name="name"
      placeholder="Full name"
      type="text"
    />
    <div
      class="styled__ErrorMessage-sc-133yddn-2 fmrEOm"
    >
        
    </div>
    <input
      class="styled__InputStyled-sc-133yddn-3 QcXMb form-control "
      name="email"
      placeholder="Email"
      type="email"
    />
    <div
      class="styled__ErrorMessage-sc-133yddn-2 fmrEOm"
    >
        
    </div>
    <input
      class="styled__InputStyled-sc-133yddn-3 QcXMb form-control "
      name="confirmEmail"
      placeholder="Confirm email"
      type="email"
    />
    <div
      class="styled__ErrorMessage-sc-133yddn-2 fmrEOm"
    >
        
    </div>
    <button
      class="styled__StyledButton-sc-1yy0gyz-0 kSrkyH"
      type="submit"
    >
      Send
    </button>
  </form>
</DocumentFragment>
`;
